ludwig/modules/fully_connected_modules.py

```diff
@@ -93,8 +92,10 @@ def __init__(
             self.layers.append(Dropout(dropout))
 
     def forward(self, inputs, mask=None):
-        hidden = inputs
+        if inputs.dtype == torch.bool:
+            inputs = inputs.to(torch.float32)
```

Comment:
Why is this necessary?

Reply:
Binary input gets passed with dtype `torch.bool` and the linear layer weights are `torch.float32` and they need to be of the same dtype.

Reply:
I suggest adding a comment to that effect to this code block

Reply:
added!

Reply:
Should we do this case at the feature / encoder level instead of down here?

Reply:
Specifically, Binary inputs only support two encoders: dense and passthrough. Both of them require a cast from bool to float32 as the first step. If we introduce a new encoder, we'll probably need to do the same thing. If we were to make this change of casting at the feature level, we could remove `BinaryPassthroughEncoder` and have binary features just use the generic `passthrough` and `dense` encoders.

Reply:
Agree that it's a lot more elegant to do it at the feature level. I made the change which required adding an `is_input_feature` argument to `add_feature_data`.

Reply:
I would rather not do it in preprocessing, but during the feature's forward function. See comment below.